### ISSUE TITLE
private image registry authentication is broken

### ISSUE THREAD

**Issue by fvigotti** [coreos/coreos-kubernetes#581]
with kubernetes 1.3.0 
- Docker version 1.10.3, build 20f81dd
- ubuntu 16.04 lts
- registry v2 ( with auth and tls, not certified ) on a public domain accessed though a public ip

I've spent two days trying to pull image from my private registry , the only solution that worked is this : #498   ( which is now closed, but in my opinion is a bug that should be opened to be fixed by someone more expert than me.. :) ) 

docs says that kubelet would use the $HOME/.dockercfg or $HOME/.docker/config.json and specify how to use the secrets specific type "kubernetes.io/dockerconfigjson" also to use : " kubectl create secret docker-registry"
I've tried every format I've seen around ( with/without prepending "https" to the url , using a json format as  
{
    "auths":
    {
    "quay.io": {
        "auth": "xXxXxXxXxXx=",
        "email": "username@example.com"
    },
}
or 
{
    "quay.io": {
        "auth": "xXxXxXxXxXx=",
        "email": "username@example.com"
}, removing newlines , 
) I've tried all possible permutations of those params,  placing it in various secrets format or node , nothing worked, errors also was misleading ( image not found, "no basic auth credentials" ) and the "no basic auth credentials" error appear when 
THERE IS an imagePullSecrets in place, in the correct namespace, and also there is a /root/.docker/config.json  and the kubelet run under root user, and the node itself can pull image without problems ( restarted everything multiple times , nothing changed )

I've also tried with 
cp /root/.docker/config.json /var/lib/kubelet/.dockercfg  ( and restart of the kubelet ) 

the only thing that worked  is : 
echo '{ "https://$PRIVATEREGPUBLICDNS:5000": {"auth": "$AUTH","email": "$EMAIL"} }' > /var/lib/kubelet/.dockercfg
with kubelet restart 

with or without "imagePullSecrets:" which never worked anyway , at least for me

shouldn't be such a pain doing a so much important thing ( I need a private registry ! ),
 the strange thing is also that authentication from registry that are not private/custom seems to work ( I've also created an account in quay.io to test this and it worked )

**Comment by cyrillk** [comment]
@fvigotti How did you solve the issue? It doesn't work for me either.. (

**Comment by preflightsiren** [comment]
I'm hitting this same issue.

**Comment by ddgenome** [comment]
I fixed it by using homebrew to check out an old version of `kubectl` and creating the secret with that.

```
$ brew switch kubernetes-cli 1.5.5
$ kubectl create secret docker-registry NAME --docker-server=https://DOCKER_REGISTRY --docker-username=DOCKER_USER --docker-email=DOCKER_EMAIL --docker-password=DOCKER_PASSWORD
```

**Comment by ghost** [comment]
In case if someone finds this useful for K8s =< 1.9.0.  ``kubernetes.io/dockercfg`` was used instead of``kubernetes.io/dockerconfigjson``. Fixed in Kubernetes 1.9.1 https://github.com/kubernetes/kubernetes/pull/57463
